crates/cdk/src/wallet/auth/auth_wallet.rs

```diff
@@ -170,89 +174,61 @@ impl AuthWallet {
 
     /// Fetch keys for mint keyset
     ///
-    /// Returns keys from local database if they are already stored.
-    /// If keys are not found locally, goes online to query the mint for the keyset and stores the [`Keys`] in local database.
+    /// Returns keys from KeyManager cache if available.
+    /// If keys are not cached, triggers a refresh and waits briefly before checking again.
     #[instrument(skip(self))]
     pub async fn load_keyset_keys(&self, keyset_id: Id) -> Result<Keys, Error> {
-        let keys = if let Some(keys) = self.localstore.get_keys(&keyset_id).await? {
-            keys
-        } else {
-            let keys = self.client.get_mint_blind_auth_keyset(keyset_id).await?;
-
-            keys.verify_id()?;
-
-            self.localstore.add_keys(keys.clone()).await?;
-
-            keys.keys
-        };
-
-        Ok(keys)
+        Ok((*self
+            .key_manager
+            .get_keys(&self.mint_url, &keyset_id)
+            .await?)
+            .clone())
     }
 
-    /// Get blind auth keysets from local database or go online if missing
+    /// Get blind auth keysets from KeyManager cache or trigger refresh if missing
     ///
-    /// First checks the local database for cached blind auth keysets. If keysets are not found locally,
-    /// goes online to refresh keysets from the mint and updates the local database.
+    /// First checks the KeyManager cache for keysets. If keysets are not cached,
+    /// triggers a refresh from the mint and waits briefly before checking again.
```

Comment:
Where is the wait mentioned in this comment I don't see it here if get_keysets does that internally we should remove the comment here